webapp/src/modules/ens/sagas.ts

```diff
@@ -148,4 +161,120 @@ export function* ensSaga() {
       yield put(claimNameCrossChainFailure(route, name, isErrorWithMessage(error) ? error.message : t('global.unknown_error')))
     }
   }
+
+  // Handle Claim name with CORAL + Credits
+  function* handleClaimNameWithCreditsRequest(action: ClaimNameWithCreditsRequestAction) {
+    const { name } = action.payload
+
+    try {
+      const wallet: Wallet | null = yield select(getWallet)
+      const credits: CreditsResponse = yield select(getCredits, wallet?.address || '')
+
+      if (!wallet) {
+        throw new Error('Wallet not connected')
+      }
+
+      if (!credits || credits.totalCredits === 0) {
+        throw new Error('No credits available')
+      }
+
+      // Get route from backend using signed fetch
+      const identity: AuthIdentity | null = yield select(getCurrentIdentity)
+      if (!identity) {
+        throw new Error('No identity available for signed fetch')
+      }
+
+      const creditsServerUrl = config.get('CREDITS_SERVER_URL')
+      const url = `${creditsServerUrl}/credits-name-route?name=${encodeURIComponent(name)}&chainId=${ChainId.MATIC_MAINNET}`
+
+      const response: Response = yield call(() =>
+        signedFetch(url, {
+          method: 'GET',
+          identity: identity
+        })
+      )
+
+      if (!response.ok) {
+        const errorData: { error: string; message: string } = yield call(() => response.json())
```

Comment:
Following the above comment, the correct way to perform a call would be to do it over the response object:
```suggestion
        const errorData: { error: string; message: string } = yield call([response, 'json')
```